src/api/web3/hooks.ts

```diff
@@ -10,8 +10,11 @@ import { createWeb3Api, getProviderByNetwork } from '.'
 // Can also be used outside by calling `createWeb3Api` directly
 export function useWeb3(): Web3 {
   const networkId = useNetworkId()
-  const provider = getProviderByNetwork(networkId)
 
   // createWeb3 caches instances per provider, safe to call multiple times
-  return useMemo(() => createWeb3Api(provider), [provider])
+  return useMemo(() => {
+    const provider = getProviderByNetwork(networkId)
+
+    return createWeb3Api(provider)
+  }, [networkId])
 }
```

Comment:
Well, my idea was to remove this file altogether :shrug:

Reply:
haha, ok, never mind!